src/conan-utils.stanza

```diff
@@ -0,0 +1,360 @@
+    add(s, ':')
+    add-all(s, package_id(cv) as String)
+    ; only use package_revision if there was a package_id
+    if package_revision(cv) != false:
+      add(s, '#')
+      add-all(s, package_revision(cv) as String)
+  to-string(s)
+
+
+;------------------------
+;--- Conan API syntax ---
+;------------------------
+;--- Search for package by name
+; curl $APIBASE/conans/search?q=pcre
+; {
+;   "results" : [ "pcre/8.45@_/_" ]
+; }
+;
+;--- Use results to search for binary packages:
+; curl $APIBASE/conans/pcre/8.45/_/_/search
+; {
+;   "139391a944851d9dacf1138cff94b3320d5775dd" : {
+;     "settings" : {
+;       "os" : "Windows",
+;       "compiler.threads" : "posix",
+;       "compiler.exception" : "seh",
+;       "arch" : "x86_64",
+;       "compiler" : "gcc",
+;       "build_type" : "Release",
+;       "compiler.version" : "11.2"
+;     },
+; (...)
+;
+; ; list revisions of a package for a specific recipe revision
+; curl -v 'http://conan.jitx.com:8081/artifactory/api/conan/conan-local/v2/conans/pcre/8.45/_/_/revisions/125d5f684fea10391ff4cbcd809a5c74/packages/6f7dea16aa97d9ca0f6b67b413781234ab190708/revisions'
+;
+;--- Use hash to get download url
+; curl $APIBASE/conans/pcre/8.45/_/_/packages/139391a944851d9dacf1138cff94b3320d5775dd/download_urls
+; {
+;   "conaninfo.txt" : "http://localhost:8082/artifactory/api/conan/conan-local/v2/files/_/pcre/8.45/_/125d5f684fea10391ff4cbcd809a5c74/package/139391a944851d9dacf1138cff94b3320d5775dd/ce6f2349e761f6350cbde62b02a687c7/conaninfo.txt",
+;   "conan_package.tgz" : "http://localhost:8082/artifactory/api/conan/conan-local/v2/files/_/pcre/8.45/_/125d5f684fea10391ff4cbcd809a5c74/package/139391a944851d9dacf1138cff94b3320d5775dd/ce6f2349e761f6350cbde62b02a687c7/conan_package.tgz",
+;   "conanmanifest.txt" : "http://localhost:8082/artifactory/api/conan/conan-local/v2/files/_/pcre/8.45/_/125d5f684fea10391ff4cbcd809a5c74/package/139391a944851d9dacf1138cff94b3320d5775dd/ce6f2349e761f6350cbde62b02a687c7/conanmanifest.txt"
+; }
+;
+;--- Use the url for "conan_package.tgz"
+; http://localhost:8082/artifactory/api/conan/conan-local/v2/files/_/pcre/8.45/_/125d5f684fea10391ff4cbcd809a5c74/package/139391a944851d9dacf1138cff94b3320d5775dd/ce6f2349e761f6350cbde62b02a687c7/conan_package.tgz
+
+doc: \<DOC>
+search for the given package name on the conan server
+@throws JSONParseException on invalid json returned
+@throws CurlException
+<DOC>
+public defn conan-search-package-name (package_name:String, -- repourl:String = DEFAULT-CONAN-URL) -> JObject :
+  var jresult
+  ; search for package name
+  val queryurl = to-string("%_/v2/conans/search?q=%_" % [repourl, package_name])
+  val headers = ["Content-Type: application/json"]
+  val curl = Curl()
+  try:
+    val response  = read-url(curl, headers, queryurl)
+    jresult = parse(response) as JObject
+  catch (e: JSONParseException):
+    free(curl)
+    throw(e)
+  catch (e:CurlException):
+    free(curl)
+    throw(e)
+  free(curl)
+  jresult
```

Comment:
You can use a `try/finally` block here: 

```
try:
    val response  = read-url(curl, headers, queryurl)
    jresult = parse(response) as JObject
finally:
  free(curl)
```

That will reduce a lot of duplication and potential for error. Further - it won't mangle the stack trace.

https://lbstanza.org/reference.html#anchor454

Reply:
It seems like this code is only used in the `deftest` - perhaps we can just remove this ?

Reply:
Good idea.  I think this was the first conan function that I wrote and didn't end up using.